lib/surface/components/link.ex

```diff
@@ -60,32 +60,27 @@ defmodule Surface.Components.Link do
   slot default
 
   def render(assigns) do
-    valid_label!(assigns)
+    unless assigns[:default] || assigns[:label] || Keyword.get(assigns.opts, :label) do
+      raise ArgumentError, "<Link /> requires a label prop or contents in the default slot"
+    end
+
     to = valid_destination!(assigns.to, "<Link />")
+    events = events_to_opts(assigns)
+    opts = link_method(assigns.method, to, assigns.opts)
+    assigns = assign(assigns, to: to, opts: events ++ opts)
 
     ~F"""
-    <a id={@id} class={@class} href={to} :attrs={to_attrs(assigns, to)}><#slot>{@label}</#slot></a>
+    <a id={@id} class={@class} href={@to} :attrs={@opts}><#slot>{@label}</#slot></a>
     """
   end
 
-  defp valid_label!(assigns) do
-    unless assigns[:default] || assigns[:label] || Keyword.get(assigns.opts, :label) do
-      raise ArgumentError, "<Link /> requires a label prop or contents in the default slot"
-    end
-  end
-
-  defp apply_method(to, method, opts) do
+  defp link_method(method, to, opts) do
     if method == :get do
       skip_csrf(opts)
     else
+      {data, opts} = Keyword.pop(opts, :data, [])
       {csrf_data, opts} = csrf_data(to, opts)
-      opts = Keyword.put_new(opts, :rel, "nofollow")
-      [data: [method: method, to: to] ++ csrf_data] ++ opts
+      [data: data ++ csrf_data ++ [method: method, to: to], rel: "nofollow"] ++ opts
```

Comment:
If data contains the `method` or the `to` keys, the `++` operator will create a keyword list with duplicated keys and it will result in invalid HTML. `Keyword.merge` will replace the previous values and solve this "potential" issue, but I admit it is less readable. WDYT?

Reply:
It turns out it's safer AND more readable 👍

Reply:
Just wanted to add that the `:attrs` directive has a safety net and only uses the first occurence if there's duplicate keys. But still good to clean it up at the component level :)